pkg/cmd/provisioning/azure/create_identity_provider.go

```diff
@@ -0,0 +1,452 @@
+	if err != nil {
+		return err
+	}
+	issuerURL, err := uploadOIDCDocuments(publicKeyPath, *storageAccount.Name, *blobContainer.Name, subscriptionID, targetDirAbsPath, ctx, cred, storageAccountKey)
+	if err != nil {
+		return err
+	}
+
+	// Create the cluster authentication object installer manifest file
+	if err = provisioning.CreateClusterAuthentication(issuerURL, targetDir); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createIdentityProviderCmd(cmd *cobra.Command, args []string) {
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	publicKeyPath := CreateIdentityProviderOpts.PublicKeyPath
+	if publicKeyPath == "" {
+		publicKeyPath = filepath.Join(CreateIdentityProviderOpts.TargetDir, provisioning.PublicKeyFile)
+	}
+
+	resourceGroupName := CreateIdentityProviderOpts.Name
+	if CreateIdentityProviderOpts.ResourceGroupName != "" {
+		resourceGroupName = CreateIdentityProviderOpts.ResourceGroupName
+	}
+
+	storageAccountName := CreateIdentityProviderOpts.Name
+	if CreateIdentityProviderOpts.StorageAccountName != "" {
+		storageAccountName = CreateIdentityProviderOpts.StorageAccountName
+	}
+
+	err = createIdentityProvider(cred,
+		CreateIdentityProviderOpts.Name,
+		CreateIdentityProviderOpts.Region,
+		resourceGroupName,
+		storageAccountName,
+		CreateIdentityProviderOpts.SubscriptionID,
+		publicKeyPath,
+		CreateIdentityProviderOpts.TargetDir,
+		CreateIdentityProviderOpts.DryRun)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// initEnvForCreateIdentityProviderCmd will ensure the destination directory exists in which to create the generated files
+func initEnvForCreateIdentityProviderCmd(cmd *cobra.Command, args []string) {
+
+	if CreateIdentityProviderOpts.StorageAccountName != "" && CreateIdentityProviderOpts.ResourceGroupName == "" {
+		log.Fatal("must provide --resource-group when --storage-account is provided")
+	}
+
+	if CreateIdentityProviderOpts.TargetDir == "" {
+		pwd, err := os.Getwd()
+		if err != nil {
+			log.Fatalf("failed to get current directory: %s", err)
+		}
+		CreateIdentityProviderOpts.TargetDir = pwd
+	}
+
+	targetDirPath, err := filepath.Abs(CreateIdentityProviderOpts.TargetDir)
+	if err != nil {
+		log.Fatalf("failed to resolve full path: %s", err)
+	}
+
+	// Create target dir if it doesn't exist
+	err = provisioning.EnsureDir(targetDirPath)
+	if err != nil {
+		log.Fatalf("failed to create target directory at %s", targetDirPath)
+	}
+
+	// Create manifests dir within the target dir if it doesn't exist
+	manifestsDir := filepath.Join(targetDirPath, provisioning.ManifestsDirName)
+	err = provisioning.EnsureDir(manifestsDir)
+	if err != nil {
+		log.Fatalf("failed to create manifests directory at %s", manifestsDir)
+	}
+}
+
+// NewCreateIdentityProviderCmd provides the "create-identity-provider" subcommand
+func NewCreateIdentityProviderCmd() *cobra.Command {
+	createIdentityProviderCmd := &cobra.Command{
+		Use:              "create-identity-provider",
```

Comment:
`"create-identity-provider"` may be overloading terms from other clouds. I'm considering renaming this subcommand to `create-oidc-issuer` or `create-oidc-storage-container` or something to that effect.

Reply:
Moving to `create-oidc-issuer`.